# Efficient amplicon panels for evaluating hybridization in the <i>Schistosome haematobium</i> species group

---

#### Roy N Platt II<sup>1</sup>, Jenna M Hulke<sup>1</sup>, Grace A Arya<sup>1</sup>, Bonnie L Webster<sup>2</sup>, Aiden M Emery<sup>2</sup>, & Timothy JC Anderson<sup>1</sup>


<p><sup>1</sup> Texas Biomedical Research Institute, San Antonio TX, United States

<sup>2</sup> Natural History Museum, London, United Kingdom</p>

Correspondence to RN Platt II (rplatt@txbiomed.org) or TJC Anderson (tanderso@txbiomed.org).

---

### Abstract
Genetic markers for detecting hybridization in nature must be informative. To screen large numbers of individuals they should also be inexpensive. Most studies of hybridization in the <i>Schistosoma haematobium</i> species group rely on either a two-marker system consisting of the mitochondrial <i>cox</i>1 and nuclear ITS markers or at the other extreme, single nucleotide variants (SNVs) from whole genome/exome sequencing.  <i>cox</i>1 and ITS studies lack precision for identifying hybrid individuals and extrapolating population genetic parameters. Whole genome sequencing is cost-prohibitive for large-scale surveillance and not always a viable option. Here, we used ancestry estimates from a whole genome data set of ~38 million SNVs genotyped in 162 <i>S. haematobium</i> and <i>S. bovis</i> sampled across Africa.  We the re-calculated ancestry estimates from 3,638 panels of 10-100,000 SNVs randomly sampled from the whole genome dataset.  We compared estimates from the panel and whole genome data to test the resolution of reduced representation sequencing in schistosomes.  We found that panels of 500 SNVs captured >99% of the population genetic information contained in the whole genome dataset by using Procrustes transformed principal component analyses and ancestry estimates (<i>r</i>Â² = 0.85). This implies that amplicon panels can be designed in a way that is scalable and provides high genetic resolution.  The costs of genotyping an individual parasite with an amplicon panel is approximately half that of whole genome sequencing and only marginally higher that Sanger sequencing <i>cox</i>1 and ITS. Single reaction amplicon panels are also easy to use, and simpler and quicker to analyze than whole genome sequence. Our results show that medium-sized amplicon panels that target genome-wide SNVs provide an effective, informative and cost-effective approach to large scale field-based surveillance for evaluating schistosome hybridization.
