{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "0b508fef-1aef-429d-9dc9-a238bbf78f41",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "from pathlib import Path\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "93834c2e-0795-40ad-8a2a-348b4b6cfa18",
   "metadata": {},
   "outputs": [],
   "source": [
    "proj_dir=\"/master/nplatt/schisto_aim_panel\"\n",
    "results_dir=f\"{proj_dir}/results\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2b1f4c80-0f39-4d7b-b19b-bccf5dd84abc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Library ID</th>\n",
       "      <th>Museum Accession Number (NHM) or Donor ID</th>\n",
       "      <th>Predicted Species</th>\n",
       "      <th>Mitchondrial Haplotype</th>\n",
       "      <th>Population Assignment</th>\n",
       "      <th>NCBI SRA Accession</th>\n",
       "      <th>Project Citation</th>\n",
       "      <th>Country</th>\n",
       "      <th>Locality</th>\n",
       "      <th>Latitude</th>\n",
       "      <th>...</th>\n",
       "      <th>Original life-cycle stage collected</th>\n",
       "      <th>Life-cycle stage sequenced</th>\n",
       "      <th>Num Read Pairs (1e6)</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Final SNV Dataset</th>\n",
       "      <th>Discordant COX1/ITS</th>\n",
       "      <th>% S. haematobium ancestry (q)</th>\n",
       "      <th>Origin</th>\n",
       "      <th>Project (Collector)</th>\n",
       "      <th>Comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>angola_cota_57</td>\n",
       "      <td>MCF03050E0612</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907442</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Source of the Cota river</td>\n",
       "      <td>-9.26186</td>\n",
       "      <td>...</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>76.6</td>\n",
       "      <td>32.96</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>angola_cota_58</td>\n",
       "      <td>MCF03050E0613</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907441</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Source of the Cota river</td>\n",
       "      <td>-9.26186</td>\n",
       "      <td>...</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>71.6</td>\n",
       "      <td>32.83</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>angola_cota_59</td>\n",
       "      <td>MCF03050E0614</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907440</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Source of the Cota river</td>\n",
       "      <td>-9.26186</td>\n",
       "      <td>...</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>66.5</td>\n",
       "      <td>25.61</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>angola_icau_60</td>\n",
       "      <td>MCF03050E0615</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907439</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Icau Wando village</td>\n",
       "      <td>-8.6451</td>\n",
       "      <td>...</td>\n",
       "      <td>miracidia</td>\n",
       "      <td>miracidia</td>\n",
       "      <td>56.7</td>\n",
       "      <td>25.73</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>angola_icau_61</td>\n",
       "      <td>MCF03050E0615</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907438</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Icau Wando village</td>\n",
       "      <td>-8.6451</td>\n",
       "      <td>...</td>\n",
       "      <td>miracidia</td>\n",
       "      <td>miracidia</td>\n",
       "      <td>75.3</td>\n",
       "      <td>30.40</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Library ID Museum Accession Number (NHM) or Donor ID Predicted Species  \\\n",
       "0  angola_cota_57                             MCF03050E0612    S. haematobium   \n",
       "1  angola_cota_58                             MCF03050E0613    S. haematobium   \n",
       "2  angola_cota_59                             MCF03050E0614    S. haematobium   \n",
       "3  angola_icau_60                             MCF03050E0615    S. haematobium   \n",
       "4  angola_icau_61                             MCF03050E0615    S. haematobium   \n",
       "\n",
       "  Mitchondrial Haplotype      Population Assignment NCBI SRA Accession  \\\n",
       "0         S. haematobium  S. haematobium (southern)        SRR11907442   \n",
       "1         S. haematobium  S. haematobium (southern)        SRR11907441   \n",
       "2         S. haematobium  S. haematobium (southern)        SRR11907440   \n",
       "3         S. haematobium  S. haematobium (southern)        SRR11907439   \n",
       "4         S. haematobium  S. haematobium (southern)        SRR11907438   \n",
       "\n",
       "  Project Citation Country                   Locality  Latitude  ...  \\\n",
       "0           Herein  Angola  Source of the Cota river   -9.26186  ...   \n",
       "1           Herein  Angola  Source of the Cota river   -9.26186  ...   \n",
       "2           Herein  Angola  Source of the Cota river   -9.26186  ...   \n",
       "3           Herein  Angola         Icau Wando village   -8.6451  ...   \n",
       "4           Herein  Angola         Icau Wando village   -8.6451  ...   \n",
       "\n",
       "  Original life-cycle stage collected Life-cycle stage sequenced   \\\n",
       "0                           cercariae                   cercariae   \n",
       "1                           cercariae                   cercariae   \n",
       "2                           cercariae                   cercariae   \n",
       "3                           miracidia                  miracidia    \n",
       "4                           miracidia                  miracidia    \n",
       "\n",
       "  Num Read Pairs (1e6) Coverage Final SNV Dataset  Discordant COX1/ITS  \\\n",
       "0                 76.6    32.96              True                  NaN   \n",
       "1                 71.6    32.83              True                  NaN   \n",
       "2                 66.5    25.61              True                  NaN   \n",
       "3                 56.7    25.73              True                  NaN   \n",
       "4                 75.3    30.40              True                  NaN   \n",
       "\n",
       "   % S. haematobium ancestry (q)  Origin  \\\n",
       "0                        100.00%    SCAN   \n",
       "1                        100.00%    SCAN   \n",
       "2                        100.00%    SCAN   \n",
       "3                        100.00%    SCAN   \n",
       "4                        100.00%    SCAN   \n",
       "\n",
       "                                 Project (Collector) Comments  \n",
       "0  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "1  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "2  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "3  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "4  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_df=pd.read_csv(f\"{proj_dir}/data/platt_et_al_2025_stable1.csv\", sep=\",\", header=0)\n",
    "info_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea619a92-2880-45d7-9496-87dc57652b7a",
   "metadata": {},
   "source": [
    "# Admixture"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "032dfa21-c7d1-4d5e-8e15-44c38e6b77a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(f\"{results_dir}/admixture\").mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(f\"{results_dir}/admixture\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "17ca71ba-d7f6-40f0-9d21-bfa0570cd1f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "rep_vcf/scan_autosomes_maf05_ld.vcf\n",
      "\t--recode-INFO-all\n",
      "\t--recode\n",
      "\t--stdout\n",
      "ei_zambia_ERR310940rebowi\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another; will always be heterozygous and is not intended to describe called alleles\">\n",
      "ted at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "r,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "conditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "llele count in genotypes, for each ALT allele, in the same order as listed\">r,Description=\"A\n",
      "ast 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "F,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Frequency, for each ALT allele, in the same order as listed\">r=A,Type=Float,Description=\"Allele \n",
      "INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "ximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "LEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "pected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Excluding individuals in 'exclude' list\n",
      "t of 163 Individualst 162 ou\n",
      "Outputting VCF file...\n",
      "After filtering, kept 381501 out of a possible 381501 Sites\n",
      "Run Time = 162.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "conda run -n popgen --live-stream bash -c \"\n",
    "    vcftools \\\n",
    "        --vcf ../prep_vcf/scan_autosomes_maf05_ld.vcf \\\n",
    "        --remove-indv margrebowiei_zambia_ERR310940 \\\n",
    "        --recode \\\n",
    "        --recode-INFO-all \\\n",
    "        --stdout > admixture.vcf\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "74f102f0-c994-42d1-9607-bb197570e023",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n",
      "urcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to admixture.log.\n",
      "t:tions in effec\n",
      "  --allow-extra-chr\n",
      "  --double-id\n",
      "  --out admixture\n",
      "  --recode 12\n",
      "  --vcf admixture.vcf\n",
      "\n",
      "MB RAM detected; reserving 515643 MB for main workspace.\n",
      "\n",
      "--vcf: 1k variants complete.\n",
      "iants complete.\n",
      "--vcf: 3k variants complete.\n",
      "--vcf: 4k variants complete.\n",
      "e.vcf: 5k variants complet\n",
      "--vcf: 6k variants complete.\n",
      "--vcf: 7k variants complete.\n",
      "--vcf: 8k variants complete.\n",
      "variants complete.\n",
      "--vcf: 10k variants complete.\n",
      "--vcf: 11k variants complete.\n",
      "omplete.2k variants c\n",
      "--vcf: 13k variants complete.\n",
      "--vcf: 14k variants complete.\n",
      "--vcf: 15k variants complete.\n",
      "-vcf: 16k variants complete.\n",
      "--vcf: 17k variants complete.\n",
      "--vcf: 18k variants complete.\n",
      "variants complete.\n",
      "--vcf: 20k variants complete.\n",
      "--vcf: 21k variants complete.\n",
      "omplete.2k variants c\n",
      "--vcf: 23k variants complete.\n",
      "--vcf: 24k variants complete.\n",
      "--vcf: 25k variants complete.\n",
      "-vcf: 26k variants complete.\n",
      "--vcf: 27k variants complete.\n",
      "--vcf: 28k variants complete.\n",
      "variants complete.\n",
      "--vcf: 30k variants complete.\n",
      "--vcf: 31k variants complete.\n",
      "omplete.2k variants c\n",
      "--vcf: 33k variants complete.\n",
      "--vcf: 34k variants complete.\n",
      "--vcf: 35k variants complete.\n",
      "-vcf: 36k variants complete.\n",
      "--vcf: 37k variants complete.\n",
      "--vcf: 38k variants complete.\n",
      "variants complete.\n",
      "--vcf: 40k variants complete.\n",
      "--vcf: 41k variants complete.\n",
      "omplete.2k variants c\n",
      "--vcf: 43k variants complete.\n",
      "--vcf: 44k variants complete.\n",
      "--vcf: 45k variants complete.\n",
      "-vcf: 46k variants complete.\n",
      "--vcf: 47k variants complete.\n",
      "--vcf: 48k variants complete.\n",
      "variants complete.\n",
      "--vcf: 50k variants complete.\n",
      "--vcf: 51k variants complete.\n",
      "omplete.2k variants c\n",
      "--vcf: 53k variants complete.\n",
      "--vcf: 54k variants complete.\n",
      "--vcf: 55k variants complete.\n",
      "-vcf: 56k variants complete.\n",
      "--vcf: 57k variants complete.\n",
      "--vcf: 58k variants complete.\n",
      "variants complete.\n",
      "--vcf: 60k variants complete.\n",
      "--vcf: 61k variants complete.\n",
      "omplete.2k variants c\n",
      "--vcf: 63k variants complete.\n",
      "--vcf: 64k variants complete.\n",
      "--vcf: 65k variants complete.\n",
      "-vcf: 66k variants complete.\n",
      "--vcf: 67k variants complete.\n",
      "--vcf: 68k variants complete.\n",
      "variants complete.\n",
      "--vcf: 70k variants complete.\n",
      "--vcf: 71k variants complete.\n",
      "omplete.2k variants c\n",
      "--vcf: 73k variants complete.\n",
      "--vcf: 74k variants complete.\n",
      "--vcf: 75k variants complete.\n",
      "-vcf: 76k variants complete.\n",
      "--vcf: 77k variants complete.\n",
      "--vcf: 78k variants complete.\n",
      "variants complete.\n",
      "--vcf: 80k variants complete.\n",
      "--vcf: 81k variants complete.\n",
      "omplete.2k variants c\n",
      "--vcf: 83k variants complete.\n",
      "--vcf: 84k variants complete.\n",
      "--vcf: 85k variants complete.\n",
      "-vcf: 86k variants complete.\n",
      "--vcf: 87k variants complete.\n",
      "--vcf: 88k variants complete.\n",
      "variants complete.\n",
      "--vcf: 90k variants complete.\n",
      "--vcf: 91k variants complete.\n",
      "omplete.2k variants c\n",
      "--vcf: 93k variants complete.\n",
      "--vcf: 94k variants complete.\n",
      "--vcf: 95k variants complete.\n",
      "-vcf: 96k variants complete.\n",
      "--vcf: 97k variants complete.\n",
      "--vcf: 98k variants complete.\n",
      "variants complete.\n",
      "--vcf: 100k variants complete.\n",
      "--vcf: 101k variants complete.\n",
      "s complete. variant\n",
      "--vcf: 103k variants complete.\n",
      "--vcf: 104k variants complete.\n",
      "ete.f: 105k variants compl\n",
      "--vcf: 106k variants complete.\n",
      "--vcf: 107k variants complete.\n",
      "--vcf: 108k variants complete.\n",
      "vcf: 109k variants complete.\n",
      "--vcf: 110k variants complete.\n",
      "--vcf: 111k variants complete.\n",
      "2k variants complete.\n",
      "--vcf: 113k variants complete.\n",
      "--vcf: 114k variants complete.\n",
      "ants complete.ri\n",
      "--vcf: 116k variants complete.\n",
      "--vcf: 117k variants complete.\n",
      "mplete.118k variants co\n",
      "--vcf: 119k variants complete.\n",
      "--vcf: 120k variants complete.\n",
      "--vcf: 121k variants complete.\n",
      "--vcf: 122k variants complete.\n",
      "--vcf: 123k variants complete.\n",
      "--vcf: 124k variants complete.\n",
      " 125k variants complete.\n",
      "--vcf: 126k variants complete.\n",
      "--vcf: 127k variants complete.\n",
      "ariants complete.\n",
      "--vcf: 129k variants complete.\n",
      "--vcf: 130k variants complete.\n",
      " complete.k variants\n",
      "--vcf: 132k variants complete.\n",
      "--vcf: 133k variants complete.\n",
      "te.cf: 134k variants comple\n",
      "--vcf: 135k variants complete.\n",
      "--vcf: 136k variants complete.\n",
      "--vcf: 137k variants complete.\n",
      "cf: 138k variants complete.\n",
      "--vcf: 139k variants complete.\n",
      "--vcf: 140k variants complete.\n",
      "k variants complete.\n",
      "--vcf: 142k variants complete.\n",
      "--vcf: 143k variants complete.\n",
      "nts complete.aria\n",
      "--vcf: 145k variants complete.\n",
      "--vcf: 146k variants complete.\n",
      "plete. 147k variants com\n",
      "--vcf: 148k variants complete.\n",
      "--vcf: 149k variants complete.\n",
      "--vcf: 150k variants complete.\n",
      "--vcf: 151k variants complete.\n",
      "--vcf: 152k variants complete.\n",
      "--vcf: 153k variants complete.\n",
      "154k variants complete.\n",
      "--vcf: 155k variants complete.\n",
      "--vcf: 156k variants complete.\n",
      "riants complete.\n",
      "--vcf: 158k variants complete.\n",
      "--vcf: 159k variants complete.\n",
      "complete.0k variants \n",
      "--vcf: 161k variants complete.\n",
      "--vcf: 162k variants complete.\n",
      "e.vcf: 163k variants complet\n",
      "--vcf: 164k variants complete.\n",
      "--vcf: 165k variants complete.\n",
      "--vcf: 166k variants complete.\n",
      "f: 167k variants complete.\n",
      "--vcf: 168k variants complete.\n",
      "--vcf: 169k variants complete.\n",
      " variants complete.\n",
      "--vcf: 171k variants complete.\n",
      "--vcf: 172k variants complete.\n",
      "ts complete.varian\n",
      "--vcf: 174k variants complete.\n",
      "--vcf: 175k variants complete.\n",
      "lete.: 176k variants comp\n",
      "--vcf: 177k variants complete.\n",
      "--vcf: 178k variants complete.\n",
      "--vcf: 179k variants complete.\n",
      "-vcf: 180k variants complete.\n",
      "--vcf: 181k variants complete.\n",
      "--vcf: 182k variants complete.\n",
      "83k variants complete.\n",
      "--vcf: 184k variants complete.\n",
      "--vcf: 185k variants complete.\n",
      "iants complete.\n",
      "--vcf: 187k variants complete.\n",
      "--vcf: 188k variants complete.\n",
      "omplete.89k variants c\n",
      "--vcf: 190k variants complete.\n",
      "--vcf: 191k variants complete.\n",
      ".-vcf: 192k variants complete\n",
      "--vcf: 193k variants complete.\n",
      "--vcf: 194k variants complete.\n",
      "--vcf: 195k variants complete.\n",
      ": 196k variants complete.\n",
      "--vcf: 197k variants complete.\n",
      "--vcf: 198k variants complete.\n",
      "variants complete.\n",
      "--vcf: 200k variants complete.\n",
      "--vcf: 201k variants complete.\n",
      "s complete. variant\n",
      "--vcf: 203k variants complete.\n",
      "--vcf: 204k variants complete.\n",
      "ete.f: 205k variants compl\n",
      "--vcf: 206k variants complete.\n",
      "--vcf: 207k variants complete.\n",
      "--vcf: 208k variants complete.\n",
      "vcf: 209k variants complete.\n",
      "--vcf: 210k variants complete.\n",
      "--vcf: 211k variants complete.\n",
      "2k variants complete.\n",
      "--vcf: 213k variants complete.\n",
      "--vcf: 214k variants complete.\n",
      "ants complete.ri\n",
      "--vcf: 216k variants complete.\n",
      "--vcf: 217k variants complete.\n",
      "mplete.218k variants co\n",
      "--vcf: 219k variants complete.\n",
      "--vcf: 220k variants complete.\n",
      "--vcf: 221k variants complete.\n",
      "--vcf: 222k variants complete.\n",
      "--vcf: 223k variants complete.\n",
      "--vcf: 224k variants complete.\n",
      " 225k variants complete.\n",
      "--vcf: 226k variants complete.\n",
      "--vcf: 227k variants complete.\n",
      "ariants complete.\n",
      "--vcf: 229k variants complete.\n",
      "--vcf: 230k variants complete.\n",
      " complete.k variants\n",
      "--vcf: 232k variants complete.\n",
      "--vcf: 233k variants complete.\n",
      "te.cf: 234k variants comple\n",
      "--vcf: 235k variants complete.\n",
      "--vcf: 236k variants complete.\n",
      "--vcf: 237k variants complete.\n",
      "cf: 238k variants complete.\n",
      "--vcf: 239k variants complete.\n",
      "--vcf: 240k variants complete.\n",
      "k variants complete.\n",
      "--vcf: 242k variants complete.\n",
      "--vcf: 243k variants complete.\n",
      "nts complete.aria\n",
      "--vcf: 245k variants complete.\n",
      "--vcf: 246k variants complete.\n",
      "plete. 247k variants com\n",
      "--vcf: 248k variants complete.\n",
      "--vcf: 249k variants complete.\n",
      "--vcf: 250k variants complete.\n",
      "--vcf: 251k variants complete.\n",
      "--vcf: 252k variants complete.\n",
      "--vcf: 253k variants complete.\n",
      "254k variants complete.\n",
      "--vcf: 255k variants complete.\n",
      "--vcf: 256k variants complete.\n",
      "riants complete.\n",
      "--vcf: 258k variants complete.\n",
      "--vcf: 259k variants complete.\n",
      "complete.0k variants \n",
      "--vcf: 261k variants complete.\n",
      "--vcf: 262k variants complete.\n",
      "e.vcf: 263k variants complet\n",
      "--vcf: 264k variants complete.\n",
      "--vcf: 265k variants complete.\n",
      "--vcf: 266k variants complete.\n",
      "f: 267k variants complete.\n",
      "--vcf: 268k variants complete.\n",
      "--vcf: 269k variants complete.\n",
      " variants complete.\n",
      "--vcf: 271k variants complete.\n",
      "--vcf: 272k variants complete.\n",
      "ts complete.varian\n",
      "--vcf: 274k variants complete.\n",
      "--vcf: 275k variants complete.\n",
      "lete.: 276k variants comp\n",
      "--vcf: 277k variants complete.\n",
      "--vcf: 278k variants complete.\n",
      "--vcf: 279k variants complete.\n",
      "-vcf: 280k variants complete.\n",
      "--vcf: 281k variants complete.\n",
      "--vcf: 282k variants complete.\n",
      "83k variants complete.\n",
      "--vcf: 284k variants complete.\n",
      "--vcf: 285k variants complete.\n",
      "iants complete.\n",
      "--vcf: 287k variants complete.\n",
      "--vcf: 288k variants complete.\n",
      "omplete.89k variants c\n",
      "--vcf: 290k variants complete.\n",
      "--vcf: 291k variants complete.\n",
      ".-vcf: 292k variants complete\n",
      "--vcf: 293k variants complete.\n",
      "--vcf: 294k variants complete.\n",
      "--vcf: 295k variants complete.\n",
      ": 296k variants complete.\n",
      "--vcf: 297k variants complete.\n",
      "--vcf: 298k variants complete.\n",
      "variants complete.\n",
      "--vcf: 300k variants complete.\n",
      "--vcf: 301k variants complete.\n",
      "s complete. variant\n",
      "--vcf: 303k variants complete.\n",
      "--vcf: 304k variants complete.\n",
      "ete.f: 305k variants compl\n",
      "--vcf: 306k variants complete.\n",
      "--vcf: 307k variants complete.\n",
      "--vcf: 308k variants complete.\n",
      "vcf: 309k variants complete.\n",
      "--vcf: 310k variants complete.\n",
      "--vcf: 311k variants complete.\n",
      "2k variants complete.\n",
      "--vcf: 313k variants complete.\n",
      "--vcf: 314k variants complete.\n",
      "ants complete.ri\n",
      "--vcf: 316k variants complete.\n",
      "--vcf: 317k variants complete.\n",
      "mplete.318k variants co\n",
      "--vcf: 319k variants complete.\n",
      "--vcf: 320k variants complete.\n",
      "--vcf: 321k variants complete.\n",
      "--vcf: 322k variants complete.\n",
      "--vcf: 323k variants complete.\n",
      "--vcf: 324k variants complete.\n",
      " 325k variants complete.\n",
      "--vcf: 326k variants complete.\n",
      "--vcf: 327k variants complete.\n",
      "ariants complete.\n",
      "--vcf: 329k variants complete.\n",
      "--vcf: 330k variants complete.\n",
      " complete.k variants\n",
      "--vcf: 332k variants complete.\n",
      "--vcf: 333k variants complete.\n",
      "te.cf: 334k variants comple\n",
      "--vcf: 335k variants complete.\n",
      "--vcf: 336k variants complete.\n",
      "--vcf: 337k variants complete.\n",
      "cf: 338k variants complete.\n",
      "--vcf: 339k variants complete.\n",
      "--vcf: 340k variants complete.\n",
      "k variants complete.\n",
      "--vcf: 342k variants complete.\n",
      "--vcf: 343k variants complete.\n",
      "nts complete.aria\n",
      "--vcf: 345k variants complete.\n",
      "--vcf: 346k variants complete.\n",
      "plete. 347k variants com\n",
      "--vcf: 348k variants complete.\n",
      "--vcf: 349k variants complete.\n",
      "--vcf: 350k variants complete.\n",
      "--vcf: 351k variants complete.\n",
      "--vcf: 352k variants complete.\n",
      "--vcf: 353k variants complete.\n",
      "354k variants complete.\n",
      "--vcf: 355k variants complete.\n",
      "--vcf: 356k variants complete.\n",
      "riants complete.\n",
      "--vcf: 358k variants complete.\n",
      "--vcf: 359k variants complete.\n",
      "complete.0k variants \n",
      "--vcf: 361k variants complete.\n",
      "--vcf: 362k variants complete.\n",
      "e.vcf: 363k variants complet\n",
      "--vcf: 364k variants complete.\n",
      "--vcf: 365k variants complete.\n",
      "--vcf: 366k variants complete.\n",
      "f: 367k variants complete.\n",
      "--vcf: 368k variants complete.\n",
      "--vcf: 369k variants complete.\n",
      " variants complete.\n",
      "--vcf: 371k variants complete.\n",
      "--vcf: 372k variants complete.\n",
      "ts complete.varian\n",
      "--vcf: 374k variants complete.\n",
      "--vcf: 375k variants complete.\n",
      "lete.: 376k variants comp\n",
      "--vcf: 377k variants complete.\n",
      "--vcf: 378k variants complete.\n",
      "--vcf: 379k variants complete.\n",
      "-vcf: 380k variants complete.\n",
      "--vcf: 381k variants complete.\n",
      "ure-temporary.bim +mporary.bed + admixt\n",
      "admixture-temporary.fam written.\n",
      "381501 variants loaded from .bim file.\n",
      "le (0 males, 0 females, 162 ambiguous) loaded from .fam.\n",
      "x .iguous sex IDs written to admixture.nose\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "s and 0 nonfounders present. 162 founder\n",
      "Calculating allele frequenc done.\n",
      "Total genotyping rate is 0.834963.\n",
      "s and 162 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "dmixture.mape.n\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "conda run -n plink --cwd . \\\n",
    "    plink \\\n",
    "        --vcf admixture.vcf \\\n",
    "        --out admixture \\\n",
    "        --double-id \\\n",
    "        --recode 12 \\\n",
    "        --allow-extra-chr\n",
    "\n",
    "conda run -n popgen --cwd . bcftools query -l admixture.vcf >samples.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "a9a7a090-ce3c-4a78-8ca8-706cda414ead",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Your job 1443404 (\"adm.k2\") has been submitted\n",
      "Your job 1443405 (\"adm.k3\") has been submitted\n",
      "Your job 1443406 (\"adm.k4\") has been submitted\n",
      "Your job 1443407 (\"adm.k5\") has been submitted\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#UNSUPERVISED \n",
    "#submit jobs to the cluster\n",
    "QSUB=\"qsub -V -cwd -S /bin/bash -q all.q \"  \n",
    "\n",
    "for K in $(seq 2 5); do\n",
    "    CMD=\"conda run -n popgen --live-stream admixture \\\n",
    "        --cv=1000 \\\n",
    "        -j12 \\\n",
    "        admixture.ped \\\n",
    "        $K \\\n",
    "        >admixutre.k$K.log\"\n",
    "\n",
    "    echo $CMD | qsub -V -cwd -S /bin/bash -q all.q  -N adm.k$K -o admixture.k$K.sge.stdout -e admixture.k$K.sge.stderr -pe smp 12\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "4e082103-bb43-4d08-8c67-9b8ce461a672",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "max_pcs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6414e0dc-a41d-40d6-aed0-ad9ede73f9f3",
   "metadata": {},
   "source": [
    "# Fst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "67d326cf-6a83-41dc-ac2c-267ed117f40f",
   "metadata": {},
   "outputs": [],
   "source": [
    "Path(f\"{results_dir}/fst\").mkdir(parents=True, exist_ok=True)\n",
    "os.chdir(f\"{results_dir}/fst\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "efd28b9b-0843-4b67-a810-3806e2404b18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Library ID</th>\n",
       "      <th>Museum Accession Number (NHM) or Donor ID</th>\n",
       "      <th>Predicted Species</th>\n",
       "      <th>Mitchondrial Haplotype</th>\n",
       "      <th>Population Assignment</th>\n",
       "      <th>NCBI SRA Accession</th>\n",
       "      <th>Project Citation</th>\n",
       "      <th>Country</th>\n",
       "      <th>Locality</th>\n",
       "      <th>Latitude</th>\n",
       "      <th>...</th>\n",
       "      <th>Original life-cycle stage collected</th>\n",
       "      <th>Life-cycle stage sequenced</th>\n",
       "      <th>Num Read Pairs (1e6)</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Final SNV Dataset</th>\n",
       "      <th>Discordant COX1/ITS</th>\n",
       "      <th>% S. haematobium ancestry (q)</th>\n",
       "      <th>Origin</th>\n",
       "      <th>Project (Collector)</th>\n",
       "      <th>Comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>angola_cota_57</td>\n",
       "      <td>MCF03050E0612</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907442</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Source of the Cota river</td>\n",
       "      <td>-9.26186</td>\n",
       "      <td>...</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>76.6</td>\n",
       "      <td>32.96</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>angola_cota_58</td>\n",
       "      <td>MCF03050E0613</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907441</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Source of the Cota river</td>\n",
       "      <td>-9.26186</td>\n",
       "      <td>...</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>71.6</td>\n",
       "      <td>32.83</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>angola_cota_59</td>\n",
       "      <td>MCF03050E0614</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907440</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Source of the Cota river</td>\n",
       "      <td>-9.26186</td>\n",
       "      <td>...</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>cercariae</td>\n",
       "      <td>66.5</td>\n",
       "      <td>25.61</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>angola_icau_60</td>\n",
       "      <td>MCF03050E0615</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907439</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Icau Wando village</td>\n",
       "      <td>-8.6451</td>\n",
       "      <td>...</td>\n",
       "      <td>miracidia</td>\n",
       "      <td>miracidia</td>\n",
       "      <td>56.7</td>\n",
       "      <td>25.73</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>angola_icau_61</td>\n",
       "      <td>MCF03050E0615</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium</td>\n",
       "      <td>S. haematobium (southern)</td>\n",
       "      <td>SRR11907438</td>\n",
       "      <td>Herein</td>\n",
       "      <td>Angola</td>\n",
       "      <td>Icau Wando village</td>\n",
       "      <td>-8.6451</td>\n",
       "      <td>...</td>\n",
       "      <td>miracidia</td>\n",
       "      <td>miracidia</td>\n",
       "      <td>75.3</td>\n",
       "      <td>30.40</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.00%</td>\n",
       "      <td>SCAN</td>\n",
       "      <td>Angola malacology survey 2013- NNTDCP/CIS and ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Library ID Museum Accession Number (NHM) or Donor ID Predicted Species  \\\n",
       "0  angola_cota_57                             MCF03050E0612    S. haematobium   \n",
       "1  angola_cota_58                             MCF03050E0613    S. haematobium   \n",
       "2  angola_cota_59                             MCF03050E0614    S. haematobium   \n",
       "3  angola_icau_60                             MCF03050E0615    S. haematobium   \n",
       "4  angola_icau_61                             MCF03050E0615    S. haematobium   \n",
       "\n",
       "  Mitchondrial Haplotype      Population Assignment NCBI SRA Accession  \\\n",
       "0         S. haematobium  S. haematobium (southern)        SRR11907442   \n",
       "1         S. haematobium  S. haematobium (southern)        SRR11907441   \n",
       "2         S. haematobium  S. haematobium (southern)        SRR11907440   \n",
       "3         S. haematobium  S. haematobium (southern)        SRR11907439   \n",
       "4         S. haematobium  S. haematobium (southern)        SRR11907438   \n",
       "\n",
       "  Project Citation Country                   Locality  Latitude  ...  \\\n",
       "0           Herein  Angola  Source of the Cota river   -9.26186  ...   \n",
       "1           Herein  Angola  Source of the Cota river   -9.26186  ...   \n",
       "2           Herein  Angola  Source of the Cota river   -9.26186  ...   \n",
       "3           Herein  Angola         Icau Wando village   -8.6451  ...   \n",
       "4           Herein  Angola         Icau Wando village   -8.6451  ...   \n",
       "\n",
       "  Original life-cycle stage collected Life-cycle stage sequenced   \\\n",
       "0                           cercariae                   cercariae   \n",
       "1                           cercariae                   cercariae   \n",
       "2                           cercariae                   cercariae   \n",
       "3                           miracidia                  miracidia    \n",
       "4                           miracidia                  miracidia    \n",
       "\n",
       "  Num Read Pairs (1e6) Coverage Final SNV Dataset  Discordant COX1/ITS  \\\n",
       "0                 76.6    32.96              True                  NaN   \n",
       "1                 71.6    32.83              True                  NaN   \n",
       "2                 66.5    25.61              True                  NaN   \n",
       "3                 56.7    25.73              True                  NaN   \n",
       "4                 75.3    30.40              True                  NaN   \n",
       "\n",
       "   % S. haematobium ancestry (q)  Origin  \\\n",
       "0                        100.00%    SCAN   \n",
       "1                        100.00%    SCAN   \n",
       "2                        100.00%    SCAN   \n",
       "3                        100.00%    SCAN   \n",
       "4                        100.00%    SCAN   \n",
       "\n",
       "                                 Project (Collector) Comments  \n",
       "0  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "1  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "2  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "3  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "4  Angola malacology survey 2013- NNTDCP/CIS and ...      NaN  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_df=pd.read_csv(f\"{proj_dir}/data/platt_et_al_2025_stable1.csv\", sep=\",\", header=0)\n",
    "info_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3aaa43d2-cf35-4aba-a8a9-6420ce4ba0f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "info_df.loc[info_df[\"Predicted Species\"] == 'S. haematobium', \"Library ID\"].to_csv(\"shae.list\", sep=\",\", header=False, index=False)\n",
    "info_df.loc[info_df[\"Predicted Species\"] == 'S. bovis', \"Library ID\"].to_csv(\"sbov.list\", sep=\",\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "7e1b2d25-6667-4756-8973-c2e0f4eb11b9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "rep_vcf/scan_autosomes_maf05_ld.vcf\n",
      "\t--weir-fst-pop shae.list\n",
      "\t--weir-fst-pop sbov.list\n",
      ".listep shae\n",
      "\t--keep sbov.list\n",
      "\n",
      "Warning: Expected at least 2 parts in FORMAT entry: ID=PGT,Number=1,Type=String,Description=\"Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another; will always be heterozygous and is not intended to describe called alleles\">\n",
      "ted at least 2 parts in FORMAT entry: ID=PID,Number=1,Type=String,Description=\"Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group\">\n",
      "r,Description=\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\">\n",
      "conditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)\">\n",
      "llele count in genotypes, for each ALT allele, in the same order as listed\">r,Description=\"A\n",
      "ast 2 parts in INFO entry: ID=AC,Number=A,Type=Integer,Description=\"Allele count in genotypes, for each ALT allele, in the same order as listed\">\n",
      "F,Number=A,Type=Float,Description=\"Allele Frequency, for each ALT allele, in the same order as listed\">\n",
      "Frequency, for each ALT allele, in the same order as listed\">r=A,Type=Float,Description=\"Allele \n",
      "INFO entry: ID=MLEAC,Number=A,Type=Integer,Description=\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "ximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\">\n",
      "LEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "pected at least 2 parts in INFO entry: ID=MLEAF,Number=A,Type=Float,Description=\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 162 out of 163 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: 0.33927\n",
      "Weir and Cockerham weighted Fst estimate: 0.54426\n",
      "After filtering, kept 381501 out of a possible 381501 Sites\n",
      "Run Time = 9.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "conda run -n popgen --live-stream \\\n",
    "    vcftools \\\n",
    "        --vcf ../prep_vcf/scan_autosomes_maf05_ld.vcf \\\n",
    "        --weir-fst-pop shae.list \\\n",
    "        --weir-fst-pop sbov.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "c9a28592-e037-4f8a-9350-8e6296f8e2b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>WEIR_AND_COCKERHAM_FST</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>124178</td>\n",
       "      <td>0.026728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>124348</td>\n",
       "      <td>0.848705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>147062</td>\n",
       "      <td>-0.015993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>147068</td>\n",
       "      <td>0.000732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>147073</td>\n",
       "      <td>0.044099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381496</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19477842</td>\n",
       "      <td>0.084907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381497</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19478168</td>\n",
       "      <td>0.973279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381498</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19478217</td>\n",
       "      <td>0.325318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381499</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19478231</td>\n",
       "      <td>0.010759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381500</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19478762</td>\n",
       "      <td>0.067064</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>381501 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              CHROM       POS  WEIR_AND_COCKERHAM_FST\n",
       "0       NC_067196.1    124178                0.026728\n",
       "1       NC_067196.1    124348                0.848705\n",
       "2       NC_067196.1    147062               -0.015993\n",
       "3       NC_067196.1    147068                0.000732\n",
       "4       NC_067196.1    147073                0.044099\n",
       "...             ...       ...                     ...\n",
       "381496  NC_067202.1  19477842                0.084907\n",
       "381497  NC_067202.1  19478168                0.973279\n",
       "381498  NC_067202.1  19478217                0.325318\n",
       "381499  NC_067202.1  19478231                0.010759\n",
       "381500  NC_067202.1  19478762                0.067064\n",
       "\n",
       "[381501 rows x 3 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fst_df=pd.read_csv(\"out.weir.fst\", sep=\"\\t\", header=0)\n",
    "fst_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "d40529de-b142-4c21-beaa-b261791f398b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>WEIR_AND_COCKERHAM_FST</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>124178</td>\n",
       "      <td>0.026728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>124348</td>\n",
       "      <td>0.848705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>147068</td>\n",
       "      <td>0.000732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>147073</td>\n",
       "      <td>0.044099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>147161</td>\n",
       "      <td>0.007621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343761</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19477842</td>\n",
       "      <td>0.084907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343762</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19478168</td>\n",
       "      <td>0.973279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343763</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19478217</td>\n",
       "      <td>0.325318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343764</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19478231</td>\n",
       "      <td>0.010759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343765</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19478762</td>\n",
       "      <td>0.067064</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>343766 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              CHROM       POS  WEIR_AND_COCKERHAM_FST\n",
       "0       NC_067196.1    124178                0.026728\n",
       "1       NC_067196.1    124348                0.848705\n",
       "2       NC_067196.1    147068                0.000732\n",
       "3       NC_067196.1    147073                0.044099\n",
       "4       NC_067196.1    147161                0.007621\n",
       "...             ...       ...                     ...\n",
       "343761  NC_067202.1  19477842                0.084907\n",
       "343762  NC_067202.1  19478168                0.973279\n",
       "343763  NC_067202.1  19478217                0.325318\n",
       "343764  NC_067202.1  19478231                0.010759\n",
       "343765  NC_067202.1  19478762                0.067064\n",
       "\n",
       "[343766 rows x 3 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fst_df = fst_df.loc[fst_df[\"WEIR_AND_COCKERHAM_FST\"]>0].reset_index(drop=True)\n",
    "fst_df.to_csv(\"fst_df.csv\", sep=\",\", header=True, index=False)\n",
    "fst_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "9e766de4-4bde-46e5-ad1b-d3bace6b321f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>WEIR_AND_COCKERHAM_FST</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2668</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>2623398</td>\n",
       "      <td>0.951922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2669</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>2662673</td>\n",
       "      <td>0.978441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2701</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>2710808</td>\n",
       "      <td>0.985495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2742</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>2736479</td>\n",
       "      <td>0.951694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2783</th>\n",
       "      <td>NC_067196.1</td>\n",
       "      <td>2753200</td>\n",
       "      <td>0.992392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343464</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19329802</td>\n",
       "      <td>0.953022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343667</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19457214</td>\n",
       "      <td>0.960472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343703</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19465685</td>\n",
       "      <td>0.957880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343729</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19470114</td>\n",
       "      <td>0.985899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343762</th>\n",
       "      <td>NC_067202.1</td>\n",
       "      <td>19478168</td>\n",
       "      <td>0.973279</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30498 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              CHROM       POS  WEIR_AND_COCKERHAM_FST\n",
       "2668    NC_067196.1   2623398                0.951922\n",
       "2669    NC_067196.1   2662673                0.978441\n",
       "2701    NC_067196.1   2710808                0.985495\n",
       "2742    NC_067196.1   2736479                0.951694\n",
       "2783    NC_067196.1   2753200                0.992392\n",
       "...             ...       ...                     ...\n",
       "343464  NC_067202.1  19329802                0.953022\n",
       "343667  NC_067202.1  19457214                0.960472\n",
       "343703  NC_067202.1  19465685                0.957880\n",
       "343729  NC_067202.1  19470114                0.985899\n",
       "343762  NC_067202.1  19478168                0.973279\n",
       "\n",
       "[30498 rows x 3 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fst_df.loc[fst_df[\"WEIR_AND_COCKERHAM_FST\"]>0.95]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "popgen",
   "language": "python",
   "name": "popgen"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
